node/src/components/storage.rs

```diff
@@ -1953,6 +1952,10 @@ fn decode_item_id<T>(raw: &[u8]) -> Result<T::Id, GetRequestError>
 where
     T: Item,
 {
+    // Transition to bytesrepr is not complete here as this function is also used in
+    // other parts of the code which we don't want to change right now. In order to
+    // fully migrate from bincode, we would have to use
+    // `bytesrepr::deserialize_from_slice` here.
     bincode::deserialize(raw).map_err(GetRequestError::MalformedIncomingItemId)
```

Comment:
Yes, ItemIDs are bincode and this is shared across networked components, like the try store. Unfortunately, the IDs are raw `Vec<u8>` without a type wrapper at the moment, and have no serialization methods. Sorry.

Reply:
The next step is to replace `bincode` with `bytesrepr` in the small network component. After that is done, I think we could change this as well.

Reply:
I think the networking component is not at fault here, but the fetcher responses and their lack of typing. I've been sitting on #2163 for too long, which should in theory fix this.